### PR TITLE
fix: admin validation broken when json is still an object

### LINKED ISSUE #20882: Can't Save content types containing JSON fields in the admin.  This doesn't match the JSON format
### Required System information

- Node.js version: v20.14.0
- NPM version: 10.8.1
- Strapi version: 5.0.0-rc.7
- Database: MySql v 8
- Operating system: OSX
- Is your project Javascript or Typescript: Typescript

### Describe the bug

After upgrading to `@strapi/strapi` version `5.0.0-rc.7`, I encountered an issue where I can no longer save content types that have JSON fields in the admin panel. The error message displayed is "This doesn't match the JSON format." However, sending the same JSON via a POST request works without any errors.

### Steps to reproduce the behavior

1. Upgrade to `@strapi/strapi` version `5.0.0-rc.7`.
2. Open the admin panel.
3. Navigate to a content type that has a JSON field.
4. Attempt to save the content type.
5. Observe the error message "This doesn't match the JSON format."

### Expected behavior

I expected to be able to save content types with JSON fields in the admin panel without encountering any errors.

### Screenshots
Screenshot 2024-07-25 at 1 47 46 PM

https://github.com/user-attachments/assets/ad10e870-05db-4611-bb30-b518ff950cca

### Code snippets
Value i'm trying to save:
{"type": "doc", "content": [{"type": "paragraph", "content": [{"text": "Lets see if this works", "type": "text"}]}]}

### PR DESCRIPTION
**Pull request by remidej**
### What does it do?

In the admin validation, handles the case where JSON content is in object form, not stringified.

### Why is it needed?

The JSON value will be an object if it was created via the content API and hasn't been modified in the input.

### How to test it?

- Create a kitchensink entry via the content API, give the json field an object value.
- Find the entry in the CM, modify another field without touching the json field
- Save. You shouldn't get a validation error.

### Related issue(s)/PR(s)

fixes #20882

### CODE REVIEW

File: packages/core/admin/admin/src/components/FormInputs/Json.tsx
Comment (remidej) [inline packages/core/admin/admin/src/components/FormInputs/Json.tsx:1]: these changes are unrelated to the fix, they're just to make the code style consistent with the rest of the repo

**Review by Marc-Roig — APPROVED** [review]
